### ISSUE TITLE
Inconsistent Github actions checkout versions

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2209]
Whilst implementing GitHub actions to aid exercising and testing Scala Native on macOS, I 
noticed that the .yml files in the mainline repository use inconsistent checkout versions:
```
$ grep checkout@ *.yml
check-cla.yml:      - uses: actions/checkout@v1
check-lint.yml:      - uses: actions/checkout@v1
run-tests.yml:      - uses: actions/checkout@v1
run-tests.yml:      - uses: actions/checkout@v2
run-tests.yml:      - uses: actions/checkout@v2
run-tests.yml:      - uses: actions/checkout@v2
run-tests.yml:      - uses: actions/checkout@v2
```
From reading the files, it appears that apparently later code uses `@v2` and apparently
historical (Scala.js origin?) code uses `@v1`.   

Now, I doubt consistency would have any functional difference but its,
as Ralph Waldo Emerson, says: "A foolish consistency is the hobgoblin of little minds,
 adored by little statesmen and philosophers and divines." To this
list I might add software engineers.

Does anybody know of a good reason I should not submit a PR to
have all three use `@v2`.  That version has been out for several
years, and is the one documented in the current GitHub docs.

I am using `checkout@v2` in my own work and have noticed
zero problems.

Please advise.

**Comment by sjrd** [comment]
We can certainly upgrade to v2 everywhere, yes.

**Comment by LeeTibbert** [comment]
Fixed via merge of PR #2215
